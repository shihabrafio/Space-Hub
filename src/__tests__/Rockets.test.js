import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore, combineReducers } from '@reduxjs/toolkit';
import Rockets from '../components/Rocket';
import rocketReducer from '../redux/rockets/rocketsSlice';

describe('Rockets component', () => {
  it('renders correctly', () => {
    const rockets = [
      {
        id: 1,
        name: 'Rocket 1',
        flickr_images: 'rocket1.jpg',
        description: 'Rocket 1 Description',
        reserved: false,
      },
      {
        id: 2,
        name: 'Rocket 2',
        flickr_images: 'rocket2.jpg',
        description: 'Rocket 2 Description',
        reserved: true,
      },
    ];

    const rootReducer = combineReducers({
      rockets: rocketReducer,
    });

    const store = configureStore({
      reducer: rootReducer,
      preloadedState: {
        rockets: {
          rockets,
          status: 'success',
          error: null,
        },
      },
    });

    const { container } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
