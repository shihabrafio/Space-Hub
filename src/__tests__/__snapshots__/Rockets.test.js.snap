// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component renders correctly 1`] = `
<div>
  <section
    class="container"
  >
    <div
      class="card"
    >
      <div
        class="img"
      >
        <img
          alt="123"
          src="rocket1.jpg"
        />
      </div>
      <div
        class="card-content"
      >
        <h3>
          Rocket 1
        </h3>
        <p
          class="desc"
        >
          Rocket 1 Description
        </p>
        <button
          class="btn primary-btn"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="img"
      >
        <img
          alt="123"
          src="rocket2.jpg"
        />
      </div>
      <div
        class="card-content"
      >
        <h3>
          Rocket 2
        </h3>
        <p
          class="desc"
        >
          <span
            class="badge"
          >
            Reserved
          </span>
          Rocket 2 Description
        </p>
        <button
          class="btn secondary-outline-btn"
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </div>
  </section>
</div>
`;
